### DISCUSSION TITLE
Bluetooth Internet Tethering Support

### DISCUSSION THREAD

**Discussion by jasmohan-narula** [micropython/micropython#15439] [Ideas]
As far as I know, microcontrollers like ESP32 has Bluetooth Classic and ESP32-S3 has Bluetooth 5.0 BLE. They should be able to support Bluetooth Tethering. But I don't think it has been implemented yet in micropython.

**Feature Description:**
Bluetooth PAN (Personal Area Network) Profile: Implementation of the Bluetooth PAN profile to enable internet tethering.
The ability for any micropython compatible board having Bluetooth Classic/BLE device (like ESP32, ESP32S3) to connect to the internet via Bluetooth tethering, utilizing an Android or other Bluetooth-enabled device as the internet source would be great. This feature would allow users to access internet connectivity on their ESP32 without relying on traditional WiFi networks, providing greater flexibility and use-case scenarios, especially in environments where WiFi is not available or practical. Example mobile IoT devices like making smartwatches.

If anyone has extensive knowledge working with micropython ESP32 SDK, please give your inputs in whether this is possible to implement in a reasonable timeframe?
Can anyone suggest some tutorials on how I can look into implementing this?

**Comment by peterhinch** [comment]
Why not set the Android device to be a WiFi hotspot and connect the ESP32x to that?

**Reply by zcattacz** [reply]
Power saving? Out of the 10 times I try BT PAN, there are 1-4 times either the Android devices or the PC needs to have BT restarted for link recovery for any serious use like watching an online video or so. It's just never too stable. 
Ping works and always works ... just with jitter...; its fixable though, the devices gotta be close, real close sometimes ...

**Reply by godevpy** [reply]
When the Android device is used as an AP, it cannot connect to other Wi-Fi at same time.

**Comment by vshymanskyy** [comment]
This is more valid for devices like nRF52

**Reply by jasmohan-narula** [reply]
If ESP32 supports it, then it can increase the useability of them for usecases like smartwatches.
WiFi consumes more power than Bluetooth. So, being able to connect to the internet with just Bluetooth can be useful.

**Comment by KnorkFox** [comment]
We all agree, that ip-service, should go over the air, not over cable or IR. We just need to select between Wifi and BT.

Wifi is already possible. 
As a beginner, i do not know a use case for a microcontroller with BT and a permanent connection into the internet. Could someone here, give us a typical example where this BT-connection into the open water of the internet is useful? 

I have asked google, on this topic. If you really need this connection, you could use an external BT-module already. It needs to be connected to your microcontroller and you have to configure it manually, once for pairing, with AT-commands.
Here is the Link: 
https://www.google.com/search?q=how+to+pair+a+microcontroller+with+bt-pan&client=ubuntu-chr&hs=1bq&sca_esv=6bae24b5c791315b&ei=ZLu1aNyGA9CF9u8Pv6i50Qc&oq=how+to+pair+a+microcontroller&gs_lp=Egxnd3Mtd2l6LXNlcnAiHWhvdyB0byBwYWlyIGEgbWljcm9jb250cm9sbGVyKgIIADIFECEYoAEyBRAhGJ8FMgUQIRifBTIFECEYnwUyBRAhGJ8FMgUQIRifBTIFECEYnwUyBRAhGJ8FMgUQIRifBTIFECEYnwVIkHdQyRJY51dwAXgBkAEAmAF8oAGeD6oBBDIwLjK4AQHIAQD4AQGYAhegAv8PwgIKEAAYsAMY1gQYR8ICBRAAGIAEwgIOEAAYgAQYsQMYgwEYigXCAgQQABgDwgIFEC4YgATCAgcQABiABBgTwgIGEAAYFhgewgIFEAAY7wXCAgcQIRigARgKmAMAiAYBkAYEkgcEMTguNaAHqoQBsgcEMTcuNbgH-w_CBwYwLjcuMTbIB04&sclient=gws-wiz-serp
